.github/workflows/test.yml

```diff
@@ -69,124 +69,99 @@ jobs:
-          restore-keys: |
-            ${{ runner.OS }}-yarn-cache-
-
-      - name: Rebuild detox
-        if: steps.cache.outputs.cache-hit == 'true'
-        run: yarn detox-setup
-
-      - name: Install Node Dependencies
-        if: steps.cache.outputs.cache-hit != 'true'
-        run: yarn --frozen-lockfile
-
       - name: Cache ruby gems
-        uses: actions/cache@v1
+        uses: actions/cache@v2
+        id: ruby_cache
         with:
           path: ios/vendor/bundle
-          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/ios/Gemfile.lock') }}
           restore-keys: |
             ${{ runner.os }}-gems-
 
-      - name: Install ruby gems (including Cocoapods)
+      - name: Install Ruby Dependencies
+        if: steps.ruby_cache.outputs.cache-hit != 'true'
         run: |
           bundle config path vendor/bundle
           bundle install --jobs 4 --retry 3
         working-directory: ./ios
 
-      - name: Cache CocoaPods
-        uses: actions/cache@v1
+      - uses: actions/cache@v2
+        id: podfile_cache
         with:
           path: ios/Pods
-          key: ${{ runner.OS }}-pods-${{ hashFiles('**/Podfile.lock') }}
+          key: ${{ runner.os }}-pods-${{ hashFiles('**/ios/Podfile.lock') }}
           restore-keys: |
-            ${{ runner.OS }}-pods-
+            ${{ runner.os }}-pods-
 
-      - run: bundle exec pod install --repo-update
+      - name: Install Cocoapods
+        if: steps.podfile_cache.outputs.cache-hit != 'true'
+        run: bundle exec pod install --repo-update
         working-directory: ./ios
 
-      - run: yarn build:e2e:ios
+      - name: Unit Test iOS
+        uses: maierj/fastlane-action@v1.4.0
+        env:
+          MATCH_PASSWORD: ${{ secrets.MATCH_PASSWORD }}
+          FASTLANE_PASSWORD: ${{ secrets.FASTLANE_PASSWORD }}
+          MATCH_GIT_BASIC_AUTH: ${{ secrets.MATCH_GIT_BASIC_AUTH }}
+          SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+        with:
+          lane: 'gps_unit_test'
+          subdirectory: 'ios'
+          bundle-install-path: 'vendor/bundle'
 
       - uses: actions/upload-artifact@v1
         with:
           name: e2e-ios-bin
-          path: ios/build/Build/Products/Release-iphonesimulator/PathCheckGPS.app
+          path: ios/fastlane/test_output/Staging-iphonesimulator/PathCheckGPS.app
+
+      - name: Upload Unit Test Results
+        if: always()
+        uses: actions/upload-artifact@v1
+        with:
+          name: JUnit Report
+          path: ios/fastlane/test_output/report.junit
+
+      - name: Upload HTML Report
+        if: always()
+        uses: actions/upload-artifact@v1
+        with:
+          name: HTML Report
+          path: ios/fastlane/test_output/report.html
 
   e2e-iphone-se:
     runs-on: macOS-latest
-    needs: e2e-ios-build
-    timeout-minutes: 15
+    needs: unit-test-ios
+    timeout-minutes: 20
```

Comment:
Oh man, do the builds really take more than 15 minutes?

Reply:
sometimes yep, 18 mins ish. We'd love to find a way to cache the build artifacts and reuse them, but with xcode I am not sure this is possible.